### PR TITLE
BaSP-TG-20: Add and edit employee page

### PR DESCRIPTION
**Pull request by JuanMornik**
- [x] Success and error messages for add and edit forms
- [x] Edit form show previous info
- [x] Add and edit forms doesn't reload the page

The styles of this page are not following Figma screenflow

### CODE REVIEW

File: src/Components/Employees/AddForm/AddEmployee.module.css [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+
+.container form div {
+  display: flex;
+  flex-direction: column;
+  width: 48%;
+}
+
+.container form div:not(:last-child) input {
+  height: 22px;
+  border: 1px solid grey;
```
Comment (EzeRocco) [inline src/Components/Employees/AddForm/AddEmployee.module.css:34]: trata de poner el color en hexadecimal. como en la linea 47

File: src/Components/Employees/index.js [hunk 2]
```diff
@@ -1,27 +1,44 @@
...
-    fetch(`https://jsonplaceholder.typicode.com/users`)
+    fetch(`${process.env.REACT_APP_API_URL}/employees`)
       .then((response) => response.json())
       .then((response) => {
-        saveEmployees(response);
+        saveEmployees(response.data);
       });
   }, []);
 
+  // const addItem = ({ firstName, lastName, phone, email, password, active }) => {
```
Comment (EzeRocco) [inline src/Components/Employees/index.js:16]: si no usas la función, acodarte de eliminar todo lo comentado

**Review by CristianAlcober — CHANGES_REQUESTED** [review]
Good job Juan! There are a couple of things that should be revised:

- [x] The 'email' and 'phone' columns have switchet parameters between them.
- [ ] The modal should close by itself after hitting the submit button, in the add or edit employee.
- [ ] The fields do not show how to correct an error.
- [ ] The fields do not show which fields have wrong data.
- [ ] Personally, I think the 'Active' fieldset should not be filled by the user, but that is just a suggestion.
I think that's pretty much the things that should get fixed, I'm sorry I did not saw this sooner.

**Comment by martumoser** [comment]
> Good job Juan! There are a couple of things that should be revised:
> 
> * [ ]  The 'email' and 'phone' columns have switchet parameters between them.
> * [ ]  The modal should close by itself after hitting the submit button, in the add or edit employee.
> * [ ]  The fields do not show how to correct an error.
> * [ ]  The fields do not show which fields have wrong data.
> * [ ]  Personally, I think the 'Active' fieldset should not be filled by the user, but that is just a suggestion.
>   I think that's pretty much the things that should get fixed, I'm sorry I did not saw this sooner.

**Review by martumoser — CHANGES_REQUESTED** [review]
La funcionalidad de edit Employee funciona correctamente pero requiere que se actualice la pagina para volver a hacer un get y visualizar esos cambios. En esta problematica buscamos manejar el manejo de data desde el state de react.En este video hay explicacion de como implementar la funcionalidad del edit haciendo uso del state de react: https://youtu.be/cByLkySU4Ho y sino podes preguntarle a tus compañeros o a mi

File: src/Components/Employees/AddForm/AddEmployee.jsx [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+          <input
+            type="password"
+            name="password"
+            value={userInput.password}
+            onChange={onChange}
+          ></input>
+        </div>
+        <div>
+          <label>Active</label>
+          <input type="text" name="active" value={userInput.active} onChange={onChange}></input>
```
Comment (LucianoAlarcon) [inline src/Components/Employees/AddForm/AddEmployee.jsx:97]: Podrias cambiar el active a un select para mejorar la UX

File: src/Components/Employees/EditForm/EditForm.jsx [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+          <input
+            type="password"
+            name="password"
+            value={userInput.password}
+            onChange={onChange}
+          ></input>
+        </div>
+        <div>
+          <label>Active</label>
+          <input type="text" name="active" value={userInput.active} onChange={onChange}></input>
```
Comment (LucianoAlarcon) [inline src/Components/Employees/EditForm/EditForm.jsx:107]: Podrias cambiar el active a un select para mejorar un poco mas la experiencia del usuario y que sea mas intuitivo

**Review by LucianoAlarcon — CHANGES_REQUESTED** [review]
Esta muy bien, falta agregar con que te paso Martu y revisa los comentarios que te deje

**Review by LucianoAlarcon — APPROVED** [review]
🚀🚀

**Review by martumoser — APPROVED** [review]
PR APPROVE

**Review by CristianAlcober — APPROVED** [review]
Excellent job Juan! Everything works splendidly, keep up the good work!

**Pull request merged**